packages/core/src/components/TextField/getMaskedValue.ts

```diff
@@ -34,4 +28,44 @@ const getMaskedValue = (event: ChangeEvent<HTMLInputElement>, mask: string): str
     return newValue;
 };
 
+export const getMaskedValue = (event: React.ChangeEvent<HTMLInputElement>, mask: string) => {
+    const { value, selectionStart } = event.target;
+    let maskedValue;
+
+    if (selectionStart && selectionStart < value.length) {
+        maskedValue = value.slice(0, selectionStart);
+    } else {
+        maskedValue = value
+            .replace(/[^a-zA-Z0-9]/g, '') //NOSONAR
+            .split('')
+            .reduce((acc: string, c: string) => applyMasking(acc + c, mask), '');
+    }
+
+    //TODO: Handle the case when the user deletes the character in the middle of the mask
```

Comment:
We can create an issue for this and add this code to the descriptions

Reply:
Done